internal/saucecloud/xcuitest.go

```diff
@@ -171,55 +236,47 @@ func (r *XcuitestRunner) startJob(jobOpts chan<- job.StartOptions, appFileID, te
 func (r *XcuitestRunner) calculateJobsCount(suites []xcuitest.Suite) int {
 	jobsCount := 0
 	for _, s := range suites {
-		jobsCount += len(s.Devices)
+		jobsCount += len(enumerateDevices(s.Devices, s.Simulators))
 	}
 	return jobsCount
 }
 
-// archiveAppsToIpaIfRequired checks if apps are a .ipa package. Otherwise, it generates one.
-func archiveAppsToIpaIfRequired(project *xcuitest.Project) error {
-	var err error
-	cache := map[string]string{}
-	project.Xcuitest.App, err = archiveAppToIpaIfRequired(project.Xcuitest.App, cache)
-	if err != nil {
-		return err
+func archive(src string, archiveType archiveType) (string, error) {
+	switch archiveType {
+	case ipaArchive:
+		return archiveAppToIpa(src)
+	case zipArchive:
+		return archiveAppToZip(src)
 	}
-
-	for i, s := range project.Suites {
-		project.Suites[i].TestApp, err = archiveAppToIpaIfRequired(s.TestApp, cache)
-		if err != nil {
-			return err
-		}
-	}
-	return nil
+	return "", fmt.Errorf("unknown archive type: %s", archiveType)
 }
 
-func archiveAppToIpaIfRequired(appPath string, cache map[string]string) (string, error) {
-	if apps.IsStorageReference(appPath) {
+func archiveAppToZip(appPath string) (string, error) {
+	if strings.HasSuffix(appPath, ".zip") {
 		return appPath, nil
 	}
 
-	if strings.HasSuffix(appPath, ".ipa") {
-		return appPath, nil
-	}
-
-	if cachedApp, ok := cache[appPath]; ok {
-		return cachedApp, nil
+	log.Info().Msgf("Archiving %s to .zip", path.Base(appPath))
+	fileName := fmt.Sprintf("%s-*.zip", strings.TrimSuffix(path.Base(appPath), ".app"))
+	tmpFile, err := os.CreateTemp(os.TempDir(), fileName)
```

Comment:
When/where is `tmpFile` deleted (it's the caller's responsibility to 🗑️ )?

Reply:
Ah, good point. Updated both archive functions to take a target dir arg.